### PR TITLE
Don't 500 when builder gets a report that's no longer editable in builder

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?177807

@czue Not at all sure if this is what you had in mind for handling this error...it isn't really much improvement over getting a 500. Won't be offended if this gets closed, figured it'd be better to get feedback than spend time going in a potentially wrong direction.

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
 )
 from corehq.apps.users.decorators import require_permission
 from corehq.apps.users.models import Permissions
+from corehq.util.context_processors import domain as domain_context
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: F401 'domain_context' imported but unused

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -220,7 +221,11 @@ def dispatch(self, request, *args, **kwargs):
                 'worker': ConfigureWorkerReport,
                 'table': ConfigureTableReport
             }[report.report_meta.builder_report_type]
-            return view_class.as_view(existing_report=report)(request, *args, **kwargs)
+            try:
+                return view_class.as_view(existing_report=report)(request, *args, **kwargs)
+            except Exception, e:
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: F841 local variable 'e' is assigned to but never used

Comment (orangejenny) [inline corehq/apps/userreports/views.py]: this does not seem to work if it's `except ApplicationNotFoundError, e`

Comment (dannyroberts) [inline corehq/apps/userreports/views.py]: `except Exception:` would fix this particular PEP8 comment. `except Exception as e:` is preferred over `except Exception, e` in recent (i.e. last 5 years) versions of python

Comment (orangejenny) [inline corehq/apps/userreports/views.py]: nm, fixed

**Comment by dannyroberts** [comment]
Possible cyclical import? Or a file not committed?
```
...
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/userreports/reports/builder/forms.py", line 44, in <module>
    from corehq.apps.userreports.reports.builder.exceptions import ApplicationNotFoundError
ImportError: No module named exceptions
```

**Comment by dannyroberts** [comment]
Ah, see you're on it. Never mind

**Comment by czue** [comment]
@orangejenny this seems basically right to me. I tweaked a little mostly to avoid sending muggles to the JSON UI. good to go from my side (will let you merge since I made last changes)

**Comment by orangejenny** [comment]
@czue ...muggles?

Tweaks look good, will merge once tests pass (small merge conflict in exceptions.py).

**Comment by orangejenny** [comment]
> @czue ...muggles?

er, nevermind :blush:

**Comment by czue** [comment]
Not a :zap: fan eh?

**Comment by orangejenny** [comment]
@dannyroberts kindly explained...hurrah for mentorship

**Pull request merged**